### ISSUE TITLE
Add support for reading sensors

### ISSUE THREAD

**Issue by OrangeGiraffe42** [buttplugio/buttplug-csharp#669]
It would open a lot of options to add some features to read various sensors.  Some basics would be an accelerometer, angle, and magnetometer, and pressure.  Is there any existing functionality for this or is it something that would be added from scratch?  I may be interested in adding this feature.

**Comment by qdot** [comment]
First off: This repo is soon to be deprecated (see readme), so any further development work would be happening in buttplugio/buttplug-rs. C# will now be an FFI layer on that, developed at buttplugio/buttplug-rs-ffi.

As for sensors, we're about to introduce our first 2 sensor readings, battery and RSSI, in v2 of the protocol spec. There are certainly plans for more sensor implementations after that, but that's spec level work, which requires extensive familiarity with the platform, its roadmap, and current/future uses, so it's a bit specialized and slow moving. There are messages proposals at https://github.com/buttplugio/buttplug/issues, and I can at add issues with frameworks for what some sensor messages might look like. Accelerometer/pressure messages are ones we've been discussing for a while, as they have applications in multiple pieces of hardware already. Angle/magnetometer might be a little much. I'm fairly frugal about what I add to the protocol, as any protocol message addition means extra maintenance for me, and it's easy to add sensor input at the application level until its proven to be needed throughout the protocol.

**Comment by OrangeGiraffe42** [comment]
Bummer, looks like it just deprecated four days ago.  I've got a Bluez device manager I was planning to ask if I should contribute once I feel it is reliable.  I wrote it to turn a Raspberry Pi 4 into a superior generic remote control via dotnet standard.  Is there a pathway for running Buttplug apps on the Raspberry Pi in the rust framework?  Rust isn't a language I'm familiar with yet.

Also, the sensor device I'm adding support in my tree is the Wit Motion WT901BLECL.  It's a pretty cool standalone accelerometer/magnetometer/inclinometer in a small battery powered package with bluetooth.  I'll probably just stick to my own branch but if there is a way I can chip in to the greater cause let me know.  I was going to define an interface for each common modality (motion, angle, etc), add a ReadSensors function passed a set of flags, and hand a generic object back, and let the caller query for interfaces for what actually was supported.

I've got no idea if magnetometer is interesting.  If muscle activity generates enough of a field to measure with it there are lots of entertaining applications.  I haven't done the math to see if that is likely to work, figured it was easier to test when I had it running.  Sensors are super important to get feedback working.

**Comment by qdot** [comment]
Yup, RPi is a Tier 2-ish platform, which mostly means "there will be builds when I remember to make them because I need to figure out how to get a container set up and onto CI for arm builds". Once I figure ARM CI builds, it'll be Tier 1. 

Right now, the way you'd access it is via getting the Rpi CLI builds (https://github.com/intiface/intiface-cli-rs, I need to make current ones) and just running that against whatever code you build. You'll be able to use C# for the app at least, but the device handling is all in rust right now. 

I maintain a cross platform BLE library in rust, and may actually have questions for you on the Bluez side. It's the platform I'm least familiar with, and the library I maintain (which right now is frakenstined from other libraries: https://github.com/deviceplug/btleplug) is written on the bluez socket interface instead of dbus, which I think may not be necessary?

I'll check out that sensor you posted about! I'm also gonna reopen and move this discussion over to https://github.com/buttplugio/buttplug, since that's sort of the standalone "roadmap" repo and it's an interesting question what might be worth it, so I can bring in other app/library devs on the discussion.
